src/Nethermind/Nethermind.Api/INethermindApi.cs

```diff
@@ -15,4 +20,15 @@ public T Config<T>() where T : IConfig
 
         (IApiWithNetwork GetFromApi, INethermindApi SetInApi) ForRpc => (this, this);
     }
+
+    public static class NethermindApiExtensions
+    {
+        public static void RegisterTxType(this INethermindApi api, TxType type, ITxDecoder decoder, ITxValidator validator)
+        {
+            ArgumentNullException.ThrowIfNull(api.TxValidator);
+
+            api.TxValidator.RegisterValidator(type, validator);
+            TxDecoder.Instance.RegisterDecoder(decoder);
+        }
+    }
```

Comment:
We might want to revisit this in the future since we'll have to add new "handlers" (ex. JSON serialization for new `TxType`s).

Reply:
this works for now, we can evolve as needed